python/pyarrow/tests/test_parquet.py

```diff
@@ -217,6 +217,21 @@ def test_no_memory_map(tempdir):
     assert table_read.equals(table)
 
 
+@pytest.mark.pandas
+def test_enable_buffered_stream(tempdir):
```

Comment:
I cannot see any easy way to test that the reading is actually buffered. We can check that the property is properly set if we store and expose the ReadProperties in the reader.